journal/pom.xml

```diff
@@ -33,11 +33,6 @@
       <artifactId>simpleclient</artifactId>
     </dependency>
 
-    <dependency>
```

Comment:
I'm confused how this works - we remove the dependency but we implemented `KryoSerializer`? How is this not causing a build problem? :thinking:

Reply:
KryoSerializer does not neeed Kryo, but it serializes via Namespaces from atomix-utils. I don't know why we needed this dependency before.

Reply:
>     1. I would rename `JournalIndexedRecord` to `RecordData`, such that a `JournalRecord` is the combination of the metadata and the data? It seems more natural to me.

:+1: 
>     2. In the `MappedJournalSegmentWriter`, we often pass the `buffer` field around as a method argument, but I'm not sure why - do we sometimes expect a different one? Not that I could find, but maybe I missed something.

:+1:

>     3. If it's possible, I'd let the serializers decide how to serialize the data for each message type directly. By this I mean, instead of writing `JournalRecordMetadataImpl`, I would let it write the metadata properties, e.g. `serializer.write(writeBuffer, checksum)`, `serializer.write(writeBuffer, index, asqn, data)`, etc. Then we only need concrete classes for reading back, not for writing. I get that it doesn't scale if we have lots of params, but it's probably fine for now? wdyt? Are these classes currently needed because we use Kryo? Even then I would consider these as implementation details of the serializer, no?

We can remove the interface and only keep the implementation classes. We are using the same implementation class for both reading and writing so there is actually no need to have an interface for it.  If we have only the implementation classes, would you then like to change the serializer api? What is the advantage of it?

Reply:
Ah! Makes sense :smile: